### DISCUSSION TITLE
[Pre-Release] v8.1.240328-rc

### DISCUSSION THREAD

**Discussion by michael-hawker** [CommunityToolkit/Windows#383] [Announcements]
## What's Changed

Welcome to the 8.1-rc release of the Windows Community Toolkit! We have a lot of new and exciting updates in this release, most notably updates for our dependent framework compatibility and a couple of long-standing controls we weren't able to migrate in the original 8.0 release, `ColorPicker` and `TabbedCommandBar`, originally contributed by @robloo and @yoshiask.

_If you're migrating from 7.x to 8.0_, check out our main [8.0 release notes here](https://github.com/CommunityToolkit/Windows/releases/tag/v8.0.230907).

You can find the `8.1.240328-rc` packages on [NuGet.org](https://www.nuget.org/profiles/Microsoft.Toolkit).

### Migration Notes 📖

We had to update the WinAppSdk TFM to `22621` to resolve incompatabilies with Win2D in the Media component when we upgraded from Windows App SDK 1.3 to 1.4. The full list of notes about this change can be found [here](https://github.com/CommunityToolkit/Tooling-Windows-Submodule/pull/161), but to summarize:
- This won't cause issues for targeting older versions of Windows, but it will require those who depend on our packages to upgrade their TFM to `net*-windows10.0.22621.0` (where star `*` is `6.0`, `7.0` [support ending soon], or `8.0`).

### Dependencies/Frameworks

We've updated our packages to support the latest .NET 8.0 (windows10.0.22621.0), Windows App SDK 1.5.240311000, and Uno 5.1.87.

* Update Microsoft.WindowsAppSDK to 1.5.240311000 by @Arlodotexe in https://github.com/CommunityToolkit/Windows/pull/369
* Add net8.0-windows to WinAppSdk by @Arlodotexe in https://github.com/CommunityToolkit/Windows/pull/328
* Updated Uno to 5.1.87 by @Arlodotexe in https://github.com/CommunityToolkit/Windows/pull/370

### New Components / Features

We've brought over the popular `ColorPicker` and `TabbedCommandBar` experiences from our old codebase, enjoy! The Camera helpers have also been fixed for use on Windows App SDK, thanks @GreenShadeZhang for helping us drive understanding on changes in the platform in this space! 🎉

`ImageCropper` has a new feature thanks to @shatyuka! @Poker-sang added a new `Spacing` property to `DockPanel`, and `GridSplitter`'s behavior for fixed columns was updated by @abdes! 🎉

* Adding ColorPicker by @niels9001 in https://github.com/CommunityToolkit/Windows/pull/241
* Porting TabbedCommandBar by @niels9001 in https://github.com/CommunityToolkit/Windows/pull/243
* fix:On the Windows app SDK using CameraHelper, FrameArrived time execution for a period of time will not be triggered. by @GreenShadeZhang in https://github.com/CommunityToolkit/Windows/pull/376
* Add overlay to ImageCropper by @shatyuka in https://github.com/CommunityToolkit/Windows/pull/364
* [GridSplitter] Fix #339: resize both items when they are fixed size to avoid the change overflowing to the irrelevant Star-sized item by @abdes in https://github.com/CommunityToolkit/Windows/pull/342
* [SettingsExpander] Support FontFamily by @niels9001 in https://github.com/CommunityToolkit/Windows/pull/333
* Enhance: Add spacing properties for `DockPanel` by @Poker-sang in https://github.com/CommunityToolkit/Windows/pull/346
* [SwitchPresenter] Sample as a loader replacement  by @niels9001 in https://github.com/CommunityToolkit/Windows/pull/285
* Adding new Segmented sample by @niels9001 in https://github.com/CommunityToolkit/Windows/pull/236

### Infrastructure

Ongoing updates/intermediary commits/improvements to the Toolkit's underlying tooling.

* Update Microsoft.WindowsAppSdk to latest stable 1.5 by @Arlodotexe in https://github.com/CommunityToolkit/Windows/pull/354
* Fixed PackageId being overridden by MSBuild.Sdk.Extras import by @Arlodotexe in https://github.com/CommunityToolkit/Windows/pull/352
* Fix CI failure after update to PowerShell 7.4.0 by @Arlodotexe in https://github.com/CommunityToolkit/Windows/pull/330
* Upgrade to C# Dev Kit by @Arlodotexe in https://github.com/CommunityToolkit/Windows/pull/353
* Upgrade to latest stable WinAppSdk 1.4 by @Arlodotexe in https://github.com/CommunityToolkit/Windows/pull/300
* Update WinAppSdk to latest 1.4.4 (1.4.231219000) by @Arlodotexe in https://github.com/CommunityToolkit/Windows/pull/314
* Cleanup project references by @Arlodotexe in https://github.com/CommunityToolkit/Windows/pull/318
* Update uno-check to 1.18.1 by @Arlodotexe in https://github.com/CommunityToolkit/Windows/pull/320
* Update Uno to 5.1.75 by @Arlodotexe in https://github.com/CommunityToolkit/Windows/pull/355
* Update to Uno 5.x by @Arlodotexe in https://github.com/CommunityToolkit/Windows/pull/311
* Updated tooling to latest main by @Arlodotexe in https://github.com/CommunityToolkit/Windows/pull/382

### Bug Fixes

Other important fixes and updates to `AdvancedCollectionView` from @Poker-sang and a Layout Cycle issue for the Segmented control by @qhy040404! 🎉

* Fix: AdvancedCollection inserts at wrong position when adding to end of source collection by @Poker-sang in https://github.com/CommunityToolkit/Windows/pull/309
* Added missing package descriptions by @Arlodotexe in https://github.com/CommunityToolkit/Windows/pull/362
* Disable Implicit animations (unsupported) when running under Uno by @Arlodotexe in https://github.com/CommunityToolkit/Windows/pull/326
* [SettingsExpander] Fix CornerRadius bug by @niels9001 in https://github.com/CommunityToolkit/Windows/pull/329
* Segmented - Update EqualPanel to avoid Layout Cycle by @qhy040404 in https://github.com/CommunityToolkit/Windows/pull/360
* Refactor: Adaptive Column Width for StaggeredLayout by @Poker-sang in https://github.com/CommunityToolkit/Windows/pull/371

## New Contributors 🎉

Thanks to all our new contributors this release for helping us to improve the Windows Community Toolkit for everyone!

* @qhy040404 made their first contribution in https://github.com/CommunityToolkit/Windows/pull/360
* @shatyuka made their first contribution in https://github.com/CommunityToolkit/Windows/pull/364
* @abdes made their first contribution in https://github.com/CommunityToolkit/Windows/pull/342
* @GreenShadeZhang made their first contribution in https://github.com/CommunityToolkit/Windows/pull/376

**Full Changelog**: https://github.com/CommunityToolkit/Windows/compare/v8.0.240109...v8.1.240328-rc

If you find any issues with this pre-release, please comment in the discussion and open an issue to let us know.

<hr /><em>This discussion was created from the release <a href='https://github.com/CommunityToolkit/Windows/releases/tag/v8.1.240328-rc'>v8.1.240328-rc</a>.</em>

**Comment by michael-hawker** [comment]
https://toolkitlabs.dev has been updated with these changes! We're aware of some issues with the new ColorPicker and TabbedCommandBar on WASM. See #385 and #321.

**Comment by lukasf** [comment]
When using SettingsCard with Uno 5.1, an exception is raised on Android and probably other platforms as well:

System.MissingMethodException: Method not found: void Microsoft.UI.Xaml.FrameworkElement.remove_IsEnabledChanged(Microsoft.UI.Xaml.DependencyPropertyChangedEventHandler)

Details and repro:

https://github.com/unoplatform/uno/issues/16073

Note: Repro uses last official WCT version, but the exception also occurs with 8.1.240328-rc

**Reply by michael-hawker** [reply]
Hmm, we've built https://toollabs.dev was built with the latest toolkit code and Uno and SettingsCard appears to work. But we don't regularly check on Android.

**Reply by lukasf** [reply]
Problem is solved by adjusting some project properties. Check here for details:

https://github.com/unoplatform/uno/issues/16073#issuecomment-2033084784

**Comment by lukasf** [comment]
Second issue I noticed: SettingsExpander is still not working on any Uno targets: The header and icons are not shown. The control can be expanded and content is shown, but the header is completely empty.

You can also see this here: https://toolkitlabs.dev/

This was also the case in all older Uno WCT builds, so it is not a new bug. But since you now officially support Uno from WCT 8, it would be great if you can look into this.

**Reply by michael-hawker** [reply]
@Arlodotexe @niels9001 I know we had talked about this before, do we have an Uno bug tracking this? Curious if it has any similar relation to the `TabbedCommandBar` issue we're seeing, though not sure if there's actual overlap in the templates there with some common element like a button.

**Comment by Poker-sang** [comment]
The `ColorPicker` still has a binding failed issue (When use `{x:Bind}` on `ColorPickerButton.SelectedColor`): 

```
CommunityToolkit.WinUI.Controls.ColorPicker
Color
Microsoft.UI.Xaml.Controls.GridView.SelectedValue
Object
Converter failed to convert value of type 'null' to type 'Color'			
```

This issue has been present since version 7.x.

**Reply by michael-hawker** [reply]
@Poker-sang is there an issue tracking this? If not, can you open one and share a minimal repro of the issue? The SelectedColor property isn't marked as nullable or anything, so I'm not sure why there'd be an issue here.

**Reply by Poker-sang** [reply]
Sure. I'll open one

**Comment by karmeye** [comment]
I got NuGet updates for `TokenizingTextBox` And `Extensions`, does that mean they were updated too, or is it just that all packages are updated to confirm to the same version number? Thx

**Reply by michael-hawker** [reply]
All packages automatically get revisioned still as part of the build. You'd have to look at the release notes for any specific changes. In general, we'll update dependencies or add support for new runtimes like .NET 8, so if you're also updating things, then it's good to just keep toolkit packages across all on the same version, even if they don't have any functional changes to components.

**Reply by karmeye** [reply]
Ok, that's what I thought, thanks!
(I was doubting whether anything had been forgotten in the release notes.)

**Reply by michael-hawker** [reply]
The release notes included in the releases area on GitHub here (which also creates these discussions now) are auto-generated by GitHub and have been pretty good lately at being all-encompassing. We do some light editing for structure and bubble up important notes above it all, but they should catch anything that has changed between the previous release.

Sometimes we'll just include the changes between preview releases, but for the stable release I try to get it to include everything since the last stable one.
